packages/core-database/lib/wallet-manager.js

```diff
@@ -349,39 +350,91 @@ module.exports = class WalletManager {
       recipient.applyTransactionToRecipient(data)
     }
 
+    this.__updateVoteBalance(sender, transaction)
+
     return transaction
   }
 
   /**
    * Remove the given transaction from a delegate.
-   * @param  {Number} type
-   * @param  {Object} data
+   * @param  {Transaction} transaction
    * @return {Transaction}
    */
-  revertTransaction ({ type, data }) {
+  revertTransaction (transaction) {
+    const { type, data } = transaction
     const sender = this.findByPublicKey(data.senderPublicKey) // Should exist
     const recipient = this.byAddress[data.recipientId]
 
     sender.revertTransactionForSender(data)
 
     // removing the wallet from the delegates index
-    if (data.type === TRANSACTION_TYPES.DELEGATE_REGISTRATION) {
+    if (type === TRANSACTION_TYPES.DELEGATE_REGISTRATION) {
       delete this.byUsername[data.asset.delegate.username]
     }
 
     if (recipient && type === TRANSACTION_TYPES.TRANSFER) {
       recipient.revertTransactionForRecipient(data)
     }
 
+    this.__updateVoteBalance(sender, transaction)
+
     return data
   }
 
+  /**
+   * Load a list of all active delegates.
+   * @param  {Number} maxDelegates
+   * @return {Array}
+   */
+  allActiveDelegates (maxDelegates) {
+    let delegates = this.allByUsername()
+
+    // NOTE: At the launch of the blockchain we may not have enough delegates.
+    // In order to have enough forging delegates we complete the list in a
+    // deterministic way (alphabetical order of publicKey).
+    if (delegates.length < maxDelegates) {
```

Comment:
it should throw if there is not enough delegates (ie genesisBlock should register at least maxDelegates)

Reply:
Should we shutdown the node in case there are not enough delegates? Doesn't seem to make sense to keep it running without enough delegates.